python/comps.py

```diff
@@ -91,11 +97,30 @@ def get_comps(
 
     # Place observations in bins. Do this in a numba-accelerated function so
     # that we can make use of fast loops
-    observation_df["price_bin"] = _bin_by_price(
-        observation_df[["id", "predicted_value"]].values,
-        price_bin_indices.values
+    observation_matrix = observation_df[["id", "predicted_value"]].values
+    taichi_obs_ndarray = ti.ndarray(dtype=int, shape=observation_matrix.shape)
+    taichi_obs_ndarray.from_numpy(observation_matrix)
+
+    price_bin_matrix = price_bin_indices.values
+    taichi_bin_ndarray = ti.ndarray(dtype=int, shape=price_bin_matrix.shape)
+    taichi_bin_ndarray.from_numpy(price_bin_matrix)
+
+    num_observations = observation_matrix.shape[0]
+    num_price_bins = price_bin_matrix.shape[0]
+
+    # Output vector
+    binned_vector = ti.ndarray(dtype=int, shape=(num_observations, 1))
```

Comment:
Taichi also cannot return fields from kernels, so we need to define the output data structure ahead of time and then mutate it in the kernel function.